### PR TITLE
updated dbt core, snowflake, and bigquery versions

### LINKED ISSUE #101: Upgrade to dbt 1.5
I noticed there was a small breaking change in where dbt 1.5 puts its built docs that we will need to respond to, but I don't think there are any major blockers.

### PR DESCRIPTION
**Pull request by britt-allen**
Fixes #101

### CODE REVIEW

**Comment by britt-allen** [comment]
IRL sigh, getting a failure: https://github.com/cagov/data-infrastructure/actions/runs/5137261032

**Comment by ian-r-rose** [comment]
If you add your branch name to the list here you can temporarily build the docs in CI to see if there are any problems: https://github.com/cagov/data-infrastructure/blob/46fb01c401df408d32bd984a7262a70767167c8f/.github/workflows/docs.yml#L5

I had run into a problem where the built docs were landing in a different place so the logic around copying them into our github pages deploy broke. Totally fixable, but annoying. But it's possible that was an unintentional regression that has since been fixed.

**Comment by ian-r-rose** [comment]
@britt-allen are you able to check/fix the docs build?

**Comment by britt-allen** [comment]
I did not realize anything needed fixing since the checks passed. Let me read more carefully what you said above and take a look. Thanks for following up @ian-r-rose

**Comment by britt-allen** [comment]
I cannot update the yaml file bc of current access token settings. Not sure how or where to change those either. I am curious to understand why this PR specifically triggers this change in the docs.yml file? @ian-r-rose

**Comment by britt-allen** [comment]
Docs build complete and ready for merging @ian-r-rose

**Comment by ian-r-rose** [comment]
Looks like the docs build had some problems, cf: https://cagov.github.io/dbt_docs_snowflake/index.html

**Comment by ian-r-rose** [comment]
This is a breaking change to where dbt places the built project in 1.5 notice where it places the docs directory when running 1.4:

https://github.com/cagov/data-infrastructure/actions/runs/5136928618/jobs/9244319759#step:9:66

vs where it places the docs directory when running 1.5:

https://github.com/cagov/data-infrastructure/actions/runs/5149087990/jobs/9271648644#step:9:64

That is what I was referring to in #101. I think it would fix things to just change the `cp` statement.

**Comment by ian-r-rose** [comment]
Found an issue for it: https://github.com/dbt-labs/dbt-core/issues/7465

Looks like it was an unintentional change and will be fixed in the next release of dbt (1.5.2). So it might fix itself if we just wait and then bump the version again.

**Comment by britt-allen** [comment]
So the current docs checks are not capable of catching this issue, correct? And that's why the checks passed despite the build actually being broken? @ian-r-rose

**Comment by britt-allen** [comment]
Should we write a test that doesn't let the check pass if the url is erroring out? I am also having a bit of trouble finding the cp statement you mentioned. May Slack you after a bit more digging. @ian-r-rose

**Comment by ian-r-rose** [comment]
> So the current docs checks are not capable of catching this issue, correct? And that's why the checks passed despite the build actually being broken? @ian-r-rose

Correct, there aren't really any docs checks right now, it's a very manual process

**Comment by ian-r-rose** [comment]
The relevant lines are here: https://github.com/cagov/data-infrastructure/blob/46fb01c401df408d32bd984a7262a70767167c8f/.github/workflows/docs.yml#L47-L54

But it does look like if we wait a week or two this will be fixed with the next dbt release.

**Comment by ian-r-rose** [comment]
Sigh, two steps forward, one step back. This is now blocked by https://github.com/dbt-labs/dbt-core/issues/7937

File: poetry.lock
Comment (britt-allen) [inline poetry.lock:1]: Hey Ian, can I get a tl;dr on why u removed some of the packages you did?

**Review by britt-allen — COMMENTED** [review]
Can't approve my own PR, but looks good.

File: poetry.lock
Comment (ian-r-rose) [reply poetry.lock:1]: These are managed by `poetry`, so I'm not sure. It's a consequence of updating the versions above: somewhere in their dependency tree something required those packages before, and no longer does.

**Comment by ian-r-rose** [comment]
Just testing the docs build on this branch, then I'll revert and if things work well, should be good to go

**Review by AeriShan-ODI — APPROVED** [review]

**Review by AeriShan-ODI — APPROVED** [review]

**Comment by AeriShan-ODI** [comment]
Tested, no issues found

**Pull request merged**